presto-main/src/main/java/com/facebook/presto/operator/index/IndexLoader.java

```diff
@@ -334,7 +347,18 @@ public boolean load(List<UpdateRequest> requests)
                 driver.updateSource(new TaskSource(sourcePlanNodeId, ImmutableSet.of(split), true));
                 while (!driver.isFinished()) {
                     ListenableFuture<?> process = driver.process();
-                    checkState(process.isDone(), "Driver should never block");
+                    long start = System.currentTimeMillis();
+                    while (!process.isDone() && System.currentTimeMillis() - start > timeout.toMillis()) {
```

Comment:
`process.get(timeout.toMillis(), TimeUnit.MILLISECONDS);` + some try/catching handling maybe?

Reply:
thanks, that's much nicer!

Reply:
Instead of using a loop, can we do:
```
try {
	process.get( 1000, MILLISECONDS);
} catch (TimeoutException e) {
    throw new PrestoException(...);
}
catch (ExecutionException e) {
    throw new PrestoException(..., e.getCause());
}
catch (InterruptedException e) {
	Thread.currentThread().interrupt();
	throw Throwables.propagate(e);
}
```